src/Psalm/Internal/Analyzer/Statements/Expression/AssertionFinder.php

```diff
@@ -1496,11 +1527,9 @@ protected static function hasLessThanCountEqualityCheck(
     }
 
     /**
-     * @param  PhpParser\Node\Expr\BinaryOp\NotIdentical|PhpParser\Node\Expr\BinaryOp $conditional
```

Comment:
Should it actually work for any binary op? E.g. `count($a) + 3`?

Reply:
It's only called with Equal, Identical, NotEqual and NotIdentical but yeah, I'll add this in doc. The old phpdoc was weird so I just deleted, I should have narrowed